include/pressio/ops/tpetra/ops_level2.hpp

```diff
@@ -226,6 +301,61 @@ product(::pressio::nontranspose mode,
   return y;
 }
 
+// -------------------------------
+// y = beta * y + alpha*A*x
+//
+// x is Pressio expression based on Eigen
+// A = tpetra::MultiVector
+// y = tpetra vector
+// -------------------------------
+template < class A_type, class x_type, class y_type, class alpha_t, class beta_t>
+::pressio::mpl::enable_if_t<
+  // level2 common constraints
+     ::pressio::Traits<A_type>::rank == 2
+  && ::pressio::Traits<x_type>::rank == 1
+  && ::pressio::Traits<y_type>::rank == 1
+  // TPL/container specific
+  && ::pressio::is_multi_vector_tpetra<A_type>::value
+  && ::pressio::is_vector_tpetra<y_type>::value
+  && ::pressio::is_expression_acting_on_eigen<x_type>::value
+  // scalar compatibility
+  && ::pressio::all_have_traits_and_same_scalar<A_type, x_type, y_type>::value
+  && std::is_convertible<alpha_t, typename ::pressio::Traits<A_type>::scalar_type>::value
+  && std::is_convertible<beta_t, typename ::pressio::Traits<A_type>::scalar_type>::value
+  && (std::is_floating_point<typename ::pressio::Traits<A_type>::scalar_type>::value
+   || std::is_integral<typename ::pressio::Traits<A_type>::scalar_type>::value)
+  >
+product(::pressio::nontranspose /*unused*/,
+	const alpha_t & alpha,
+	const A_type & A,
+	const x_type & x,
+	const beta_t & beta,
+	y_type & y)
+{
+  const auto A_h = A.getLocalViewHost(Tpetra::Access::ReadOnlyStruct());
+  const auto y2_h = y.getLocalViewHost(Tpetra::Access::ReadWriteStruct());
+  const auto y_h = Kokkos::subview(y2_h, Kokkos::ALL(), 0);
+
+  assert( ::pressio::ops::extent(y_h, 0) == ::pressio::ops::extent(A_h, 0) );
+  assert( ::pressio::ops::extent(x, 0) == ::pressio::ops::extent(A_h, 1) );
+
+  const auto zero = ::pressio::utils::Constants<beta_t>::zero();
+  if (beta == zero) ::pressio::ops::set_zero(y_h);
+  else ::pressio::ops::scale(y_h, beta);
+  if (alpha == zero) return;
```

Comment:
minor comment but from now on, can you please write these as: I like to have braces for readability 
```
const auto zero = ::pressio::utils::Constants<beta_t>::zero();
  if (beta == zero){ 
    ::pressio::ops::set_zero(y_h);
  }
  else{ 
    ::pressio::ops::scale(y_h, beta);
  }
  if (alpha == zero){ 
    return;
   }
```

Reply:
ok, fixed on b4d7832b8b1f7bb4b120acedbad85454936b8f40

PS. we have #116 open for auto-formatting,
and now `clang-format` 15 can do `InsertBraces: true` and [other cool things](https://clang.llvm.org/docs/ClangFormatStyleOptions.html)